pkg/kv/bulk/sst_batcher.go

```diff
@@ -1001,6 +988,30 @@ func createSplitSSTable(
 	if err != nil {
 		return nil, nil, err
 	}
+	if !split {
+		return nil, nil, errors.Newf("split key %s after last key %s", last.Next())
+	}
 	right = &sstSpan{start: first, end: last.Next(), sstBytes: sstFile.Data()}
 	return left, right, nil
 }
+
+func addStatsToSplitTables(left, right, original *sstSpan, sendStartTimestamp time.Time) error {
```

Comment:
I might put a small comment on the strategy here of computing the stats for one side and reusing the original stats to calculate the remainder.